### DISCUSSION TITLE
Grouping by different value?

### DISCUSSION THREAD

**Discussion by bzbetty** [TanStack/table#4310] [Q&A] [answered]
I have a table with a due date column, I want to enable grouping by this column however not by the due date itself.  I want to make the grouping be based on week due (eg date of the start of the week for due date).

is there a way to have a different value for grouping than the accessor for the column?

**Comment by tannerlinsley** [comment]
There is a grouping value option on the column that might be what you’re looking for

**Reply by bzbetty** [reply]
I saw that but was expecting something on the columnDef for it to be usable.  Is this a matter of somehow overriding how the grouping model is created?

**Comment by bzbetty** [comment, marked as answer]
Found a solution, bit of a hack but it worked.

Added a hidden column for the value I actually wanted to group by, then switched out the button that did the grouping to one that grouped by a columnId metadata on the original column.
